src/models/base_model.rs

```diff
@@ -154,14 +230,20 @@ impl BaseModel {
         let validity_period = model
             .calculate_validity_period()
             .map_err(|_| BadModel::NoDataset)?;
+
         if validity_period.0 > validity_period.1 {
             return Err(BadModel::StartDateAfterEndDate);
         }
+        // Associate stop_points with path way
+        let association_stop_point_to_pathways =
+            Self::associate_stop_points_with_pathway(&model).unwrap();
```

Comment:
We don't want to crash the program in the Error case, so we should bubble up the error with something like
```suggestion
            Self::associate_stop_points_with_pathway(&model)
            .map_err(
                |err| BadModel::NewBadModelCase(err)
            )?;
```